v-next/hardhat/src/internal/builtin-plugins/network-manager/hook-handlers/hre.ts

```diff
@@ -15,6 +17,9 @@ export default async (): Promise<Partial<HardhatRuntimeEnvironmentHooks>> => ({
 
         if (networkManager === undefined) {
           networkManager = new NetworkManagerImplementation(
+            hre.globalOptions.network !== undefined
+              ? hre.globalOptions.network
+              : DEFAULT_NETWORK_NAME,
```

Comment:
Before the changes to the default network this was:
```
hre.globalOptions.network !== undefined
              ? hre.globalOptions.network
              : hre.config.defaultNetwork,
```
The default network PR removed the globalOptions along with the `hre.config.defaultNetwork`.

https://github.com/NomicFoundation/hardhat/pull/6974/files#diff-03173af1ae628b27bdff9793dc3e766d0a0a537dd3cd74c19c00f5934261e5bc